src/platform_impl/macos/view.rs

```diff
@@ -50,38 +50,75 @@ impl Default for CursorState {
     }
 }
 
+enum ImeState {
+    Disabled,
+    Enabled,
+    Preedit,
+    WillCommit { text: String },
+}
+
 pub(super) struct ViewState {
     ns_window: id,
+    ime_subview: IdRef,
     pub cursor_state: Arc<Mutex<CursorState>>,
     ime_spot: Option<(f64, f64)>,
     raw_characters: Option<String>,
     pub(super) modifiers: ModifiersState,
     tracking_rect: Option<NSInteger>,
+    ime_state: ImeState,
+    input_source: String,
+
+    /// True iff the application wants IME events.
+    ///
+    /// Can be set using `set_ime_enabled`
+    app_enabled_ime: bool,
+
+    /// The characters for `ReceivedCharacter` that we need to send
+    /// after sending the corresponding KeyboardInput
+    characters_to_send: Vec<char>,
 }
 
 impl ViewState {
     fn get_scale_factor(&self) -> f64 {
         (unsafe { NSWindow::backingScaleFactor(self.ns_window) }) as f64
     }
+    fn is_ime_enabled(&self) -> bool {
+        match self.ime_state {
+            ImeState::Disabled => false,
+            _ => true,
+        }
+    }
 }
 
-pub fn new_view(ns_window: id) -> (IdRef, Weak<Mutex<CursorState>>) {
+pub fn new_view(ns_window: id) -> (IdRef, IdRef, Weak<Mutex<CursorState>>) {
     let cursor_state = Default::default();
     let cursor_access = Arc::downgrade(&cursor_state);
+    let input_source = unsafe { current_input_source() };
     let state = ViewState {
         ns_window,
+        ime_subview: IdRef::new(nil),
         cursor_state,
         ime_spot: None,
         raw_characters: None,
         modifiers: Default::default(),
         tracking_rect: None,
+        ime_state: ImeState::Disabled,
+        input_source,
+        app_enabled_ime: false, // TODO: add example to show how to use IME
+        characters_to_send: Vec::with_capacity(8),
     };
     unsafe {
         // This is free'd in `dealloc`
         let state_ptr = Box::into_raw(Box::new(state)) as *mut c_void;
         let ns_view: id = msg_send![VIEW_CLASS.0, alloc];
+        let ime_subview: id = msg_send![IME_SUBVIEW_CLASS.0, alloc];
+        let ime_subview: id = msg_send![ime_subview, initWithWinit: state_ptr];
```

Comment:
Would be nice to have a comment why do we need a subview for that?

Reply:
I'll add a comment to `IME_SUBVIEW_CLASS`